File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        int max =  Integer.MIN_VALUE ;
        int cnt[] = new int[1];
        Travel(root,max,cnt);
        return cnt[0];
    }
    
private static void Travel(TreeNode root, int max , int[] cnt ){
   if(root==null)  return;
   if(root.val>= max) cnt[0]++;
   
   max = Math.max( root.val , max );
   Travel(root.left , max , cnt );
   Travel(root.right , max , cnt );

}

}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.js
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    rightSideView(root) {
        const ans = [];
        if (!root) return ans;

        const q = [root];

        while (q.length) {
            const size = q.length;

            for (let i = 0; i < size; i++) {
                const node = q.shift();

                if (node.left) q.push(node.left);
                if (node.right) q.push(node.right);

                if (i === size - 1) {
                    ans.push(node.val);
                }
            }
        }

        return ans;
    }
}